### PR TITLE
Add fields to keep metadata for Fulfillment model

### LINKED ISSUE #4865: Add Fulfilment metdata
Fulfilment model is missing metadata.

### PR DESCRIPTION
**Pull request by szewczykmira**
Closes #4865 

Also, I'm not sure if I've used proper permissions - any comments are welcome.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 319a4cad9dd2d9c847fecd539acb90cb178b7e2d (mirumee/saleor @ feature/fulfilment-metadata)
Base comparison is 069b8b82935c61cf341192108c81751a0363b907.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/order/mutations/fulfillments.py [hunk 1]
```diff
@@ -44,6 +48,38 @@ class FulfillmentCancelInput(graphene.InputObjectType):
...
+        model = models.Fulfillment
+        permission = "orders.manage_orders"
+        public = True
+
+
+class FulfillmentUpdateMeta(UpdateMetaBaseMutation):
+    class Meta:
+        description = "Updates meta for fulfillment"
+        model = models.Fulfillment
+        permission = "orders.manage_orders"
```
Comment (korycins) [inline saleor/graphql/order/mutations/fulfillments.py]: We need to confirm that basic meta update/clear should be protected by staff's permissions.

File: saleor/graphql/order/mutations/fulfillments.py [hunk 2]
```diff
@@ -44,6 +48,38 @@ class FulfillmentCancelInput(graphene.InputObjectType):
     restock = graphene.Boolean(description="Whether item lines are restocked.")
 
 
+class FulfillmentClearMeta(ClearMetaBaseMutation):
+    class Meta:
+        description = "Clears meta for fulfillment"
```
Comment (maarcingebala) [inline saleor/graphql/order/mutations/fulfillments.py]: I would slightly change it:
```suggestion
        description = "Clears metadata for fulfillment."
```
And similarly for the rest of the descriptions.

Comment (NyanKiyoshi) [reply saleor/graphql/order/mutations/fulfillments.py]: Metadata or meta-data?

File: CHANGELOG.md [hunk 3]
```diff
@@ -100,6 +100,9 @@ All notable, unreleased changes to this project will be documented in this file.
 - Fix ProductVariant with duplicated AttributeValues - #4829 by @fowczarek
 - Saleor can now be part of Apollo Federation, making seamless GraphQL schema extension possible - #4825 by @salwator
 - Added the possibility to customize the sender address and name from the dashboard - #4820 by @NyanKiyoshi
+- Add Fulfilment metadata - # @szewczykmira
+- Add Fulfillment metadata - # @szewczykmira
+- Add Fulfillment metadata - #4866 @szewczykmira
```
Comment (maarcingebala) [inline CHANGELOG.md]: We should only keep this entry (it's duplicated now):
```suggestion
- Add Fulfillment metadata - #4866 by @szewczykmira
```

**Pull request merged**